source/common.h

```diff
@@ -65,7 +65,7 @@ struct source_line
 {
     std::string text;
 
-    enum class category { empty, preprocessor, comment, import, cpp1, cpp2, rawstring };
+    enum class category : std::uint8_t { empty, preprocessor, comment, import, cpp1, cpp2, rawstring };
```

Comment:
Why is this suggested by clang-tidy? Is this meant to optimize for size? If that is the case it looks a bit like over-optimization.
It might be personal taste but the original seems nicer to read.

Reply:
Yes, it is meant to optimize for size.  https://clang.llvm.org/extra/clang-tidy/checks/performance/enum-size.html

Reply:
It doesn't make an impact in any of the places I put it, I can turn the lint off and undo these changes if you'd prefer it that way.
I tried to keep as many of the checks enabled, only turning off the most noisy ones.
I'm totally ok with turning this off as well.